Xamarin.Forms.Platform.Android/Extensions/ToolbarExtensions.cs

```diff
@@ -40,37 +41,42 @@ public static void DisposeMenuItems(this AToolbar toolbar, IEnumerable<ToolbarIt
 			foreach (var menuItem in menuItemsCreated)
 				menuItem.Dispose();
 
+			foreach (var toolbarItem in toolbarItemsCreated)
+				toolbarItem.PropertyChanged -= toolbarItemChanged;
+
 			menuItemsCreated.Clear();
+			toolbarItemsCreated.Clear();
 
-			int i = 0;
 			foreach (var item in sortedToolbarItems)
 			{
-				UpdateMenuItem(toolbar, context, menuItemsCreated, item, tintColor, toolbarItemChanged, null, updateMenuItemIcon);
-				i++;
+				UpdateMenuItem(toolbar, item, null, context, tintColor, toolbarItemChanged, menuItemsCreated, toolbarItemsCreated, updateMenuItemIcon);
 			}
 		}
 
-		internal static void UpdateMenuItem(
-			AToolbar toolbar,
-			Context context, 
-			List<IMenuItem> menuItemsCreated, 
-			ToolbarItem item, 
+		internal static void UpdateMenuItem(AToolbar toolbar,
+			ToolbarItem item,
+			int? menuItemIndex,
+			Context context,
 			Color? tintColor,
 			PropertyChangedEventHandler toolbarItemChanged,
-			int? menuItemIndex,
-			Action<Context, IMenuItem, ToolbarItem> updateMenuItemIcon = null)
+			List<IMenuItem> menuItemsCreated,
+			List<ToolbarItem> toolbarItemsCreated,
+			Action<Context,
+			IMenuItem, 
+			ToolbarItem> updateMenuItemIcon = null)
 		{
 			IMenu menu = toolbar.Menu;
 			item.PropertyChanged -= toolbarItemChanged;
 			item.PropertyChanged += toolbarItemChanged;
 
-			IMenuItem menuitem = null;
+			toolbarItemsCreated?.Add(item);
```

Comment:
I think this change needs to be moved down to

https://github.com/xamarin/Xamarin.Forms/pull/9485/files#diff-b461663043f6182b652f5aeb4b221297R79

If you test with Issue9360 and click to change the icon you'll see toolbarItemsCreated keeps getting items added that it already contains

Reply:
I had a doubt on this one, I did not understand well the menuItemIndex == null.
But now it's logic, it is if the item is in the collection or not.